subsys/bluetooth/host/scan.c

```diff
@@ -1970,8 +1970,10 @@ int bt_le_per_adv_sync_create(const struct bt_le_per_adv_sync_param *param,
 	}
 
 	if (param->options & BT_LE_PER_ADV_SYNC_OPT_FILTER_DUPLICATE) {
-		cp->options |=
-			BT_HCI_LE_PER_ADV_CREATE_SYNC_FP_FILTER_DUPLICATE;
+		if (BT_FEAT_LE_PER_ADV_ADI_SUPP(bt_dev.le.features)) {
+			cp->options |=
+				BT_HCI_LE_PER_ADV_CREATE_SYNC_FP_FILTER_DUPLICATE;
+		}
```

Comment:
If this is the case, then I think we should reject the request early. 

At the top of this function we have a similar check

```c
	if (!BT_FEAT_LE_EXT_PER_ADV(bt_dev.le.features)) {
		return -ENOTSUP;
	}
```

We should not silently ignore parameters from the application, and we should perform parameter checks before attempting to allocate buffers and PA sync objects, so I suggest to move the check to above (but keep the assignment here)

Reply:
> We should not silently ignore parameters from the application

Agree to this. Application can check and use the correct parameters.

> This will make sample failed on chipset that do not support this feature.

Fix the sample to use the feature if supported and document it accordingly.